File: samples/java_layout/src/main/java/io/customer/android/sample/java_layout/ui/inline/SectionsPagerAdapter.java
package io.customer.android.sample.java_layout.ui.inline;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import io.customer.android.sample.java_layout.ui.inline.compose.ComposeInlineExampleFragment;

public class SectionsPagerAdapter extends FragmentStateAdapter {

    public SectionsPagerAdapter(@NonNull FragmentManager fragmentManager, @NonNull Lifecycle lifecycle) {
        super(fragmentManager, lifecycle);
    }

    @Override
    public int getItemCount() {
        return 4;
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position) {
            case 0:
                return SingleInlineFragment.newInstance();
            case 1:
                return MultipleInlineFragment.newInstance();
            case 2:
                return ComposeInlineExampleFragment.newInstance();
            case 3:
                return RecyclerViewInlineFragment.newInstance();
            default:
                throw new IllegalArgumentException("Invalid position: " + position);
        }
    }
}

File: samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/ui/inline/ui/main/SectionsPagerAdapter.kt
package io.customer.android.sample.kotlin_compose.ui.inline.ui.main

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import io.customer.android.sample.kotlin_compose.ui.inline.components.MultipleInlineFragment
import io.customer.android.sample.kotlin_compose.ui.inline.components.SingleInlineFragment
import io.customer.android.sample.kotlin_compose.ui.inline.compose.KotlinComposeFragment
import io.customer.android.sample.kotlin_compose.ui.inline.recyclerview.RecyclerViewFragment

class SectionsPagerAdapter(
    fragmentManager: FragmentManager,
    lifecycle: Lifecycle
) : FragmentStateAdapter(fragmentManager, lifecycle) {

    override fun getItemCount(): Int {
        return 4
    }

    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> SingleInlineFragment.newInstance()
            1 -> MultipleInlineFragment.newInstance()
            2 -> KotlinComposeFragment.newInstance()
            3 -> RecyclerViewFragment.newInstance()
            else -> throw IllegalArgumentException("Invalid position")
        }
    }
}
